Bottom: 2aaee49d2240e2e00474d5faf72562a10bc63925
Top:    af7d3834712f69cd942c4ed0de3ad942abd8db78
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-09 01:59:06 +0100

Filter out double clicks on the layer selector widget

This makes things behave a little nicer if you click to toggle a layer's
visibility twice in quick succession. Previously, this would be
interpreted as a double-click, causing an extra synthetic click event
which would toggle the visibility an extra time.


---

diff --git a/src/hid/gtk/ghid-layer-selector.c b/src/hid/gtk/ghid-layer-selector.c
index a2bb3ff..639e2be 100644
--- a/src/hid/gtk/ghid-layer-selector.c
+++ b/src/hid/gtk/ghid-layer-selector.c
@@ -193,8 +193,11 @@ button_press_cb (GHidLayerSelector *ls, GdkEventButton *event)
   GtkTreeViewColumn *column;
   GtkTreePath *path;
 
-  /* Ignore all but left-clicks */
-  if (event->button != 1)
+  /* Ignore the synthetic presses caused by double and tripple clicks, and
+   * also ignore all but left-clicks
+   */
+  if (event->type != GDK_BUTTON_PRESS ||
+      event->button != 1)
     return TRUE;
 
   if (gtk_tree_view_get_path_at_pos (GTK_TREE_VIEW (ls),
